public/components/integrations/components/setup_integration.tsx

```diff
@@ -231,16 +296,64 @@ export function SetupBottomBar({
             isLoading={loading}
             onClick={async () => {
               setLoading(true);
-              const template = await findTemplate(integration.name);
-              await addIntegrationRequest(
-                false,
-                integration.name,
-                config.displayName,
-                template,
-                setToast,
-                config.displayName,
-                config.connectionDataSource
-              );
+              // Not sure why if I make an incrementer it doesn't change the prop,
+              // But since it doesn't, we track our progress manually.
```

Comment:
could you explain a bit what's an incrementer?

Reply:
If I make a helper `const incrementProgress = () => setProgress(progress + 1)`, it doesn't correctly identify when progress updates, so it always computes `0 + 1`. Similar situation if I use `setProgress(progress + 1)` directly. The workaround is to manually track progress locally and use that instead of the actual progress variable.

Reply:
use `setProgress((progress) => progress + 1)`. it's a bit of counterintuitive issue in react where the value of progress is saved in the function